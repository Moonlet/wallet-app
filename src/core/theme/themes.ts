export enum Theme {
    dark = 'dark',
    light = 'light'
}
