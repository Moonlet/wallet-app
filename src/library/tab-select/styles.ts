import { StyleSheet } from 'react-native';
import { ITheme } from '../../core/theme/itheme';
import { BASE_DIMENSION } from '../../styles/dimensions';

export default (theme: ITheme) =>
    StyleSheet.create({
        container: {
            marginHorizontal: BASE_DIMENSION * 3,
            borderWidth: 1,
            borderColor: theme.colors.accent,
            flexDirection: 'row'
        },
        tabButton: {
            flexBasis: 0,
            flex: 1,
            alignItems: 'center',
            justifyContent: 'center',
            paddingVertical: BASE_DIMENSION
        },
        tabButtonText: {
            fontSize: 13,
            lineHeight: 18
        },
        tabButtonSelected: {
            backgroundColor: theme.colors.accent
        },
        tabButtonTextSelected: {
            color: theme.colors.appBackground
        }
    });
