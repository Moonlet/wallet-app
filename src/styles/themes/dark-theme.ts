import { ITheme } from '../../core/theme/itheme';
import { COLORS } from '../colors';

export const darkTheme: ITheme = {
    dark: true,
    shadowGradient: ['#00000000', '#000000CC'],

    fontSize: {
        small: 12,
        regular: 16,
        large: 34
    },

    colors: {
        primary: COLORS.GRAY,
        accent: COLORS.AQUA,
        text: COLORS.WHITE,
        textDarker: COLORS.LIGHT_GRAY,
        positive: COLORS.HOT_GREEN,
        negative: COLORS.HOT_RED,

        cardBackground: COLORS.DARK_GRAY,
        appBackground: COLORS.COD_GRAY,
        settingsDivider: COLORS.BLACK
    }
};
