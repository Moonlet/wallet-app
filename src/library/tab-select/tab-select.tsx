import React from 'react';
import { TouchableOpacity, View } from 'react-native';
import { Text } from '../index';

import stylesProvider from './styles';
import { withTheme } from '../../core/theme/with-theme';
import { BORDER_RADIUS } from '../../styles/dimensions';

export interface ITabSelectProps {
    options: any;
    selected: any;
    onSelectionChange: (val: any) => void;
    styles: ReturnType<typeof stylesProvider>;
}
export const TabSelectComponent = (props: ITabSelectProps) => (
    <View style={props.styles.container}>
        {Object.keys(props.options).map((key, index) => (
            <TouchableOpacity
                key={index}
                onPress={() => {
                    props.onSelectionChange(key);
                }}
                style={[
                    props.styles.tabButton,
                    props.selected === key && props.styles.tabButtonSelected,
                    {
                        borderTopLeftRadius: index === 0 ? BORDER_RADIUS / 2 : 0,
                        borderBottomLeftRadius: index === 0 ? BORDER_RADIUS / 2 : 0,
                        borderTopRightRadius:
                            index === Object.keys(props.options).length - 1 ? BORDER_RADIUS / 2 : 0,
                        borderBottomRightRadius:
                            index === Object.keys(props.options).length - 1 ? BORDER_RADIUS / 2 : 0
                    }
                ]}
            >
                <Text
                    style={[
                        props.styles.tabButtonText,
                        props.selected === key && props.styles.tabButtonTextSelected
                    ]}
                >
                    {props.options[key].title}
                </Text>
            </TouchableOpacity>
        ))}
    </View>
);

export const TabSelect = withTheme(stylesProvider)(TabSelectComponent);
