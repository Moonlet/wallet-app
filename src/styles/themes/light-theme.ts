import { ITheme } from '../../core/theme/itheme';
import { darkTheme } from './dark-theme';

export const lightTheme: ITheme = darkTheme;
