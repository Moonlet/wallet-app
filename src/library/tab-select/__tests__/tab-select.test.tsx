import React from 'react';
import { TabSelectComponent, ITabSelectProps } from '../tab-select';
import { darkTheme } from '../../../styles/themes/dark-theme';
import styleProvider from '../styles';

import { shallow } from 'enzyme';

const props: ITabSelectProps = {
    styles: styleProvider(darkTheme),
    options: {
        '1': 'a',
        '2': 'b',
        '3': 'c'
    },
    selected: '2',
    onSelectionChange: jest.fn()
};

describe('library checkbox', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<TabSelectComponent {...props} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
