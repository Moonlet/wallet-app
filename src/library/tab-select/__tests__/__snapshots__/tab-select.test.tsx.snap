// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`library checkbox renders correctly 1`] = `
"<View style={{...}}>
  <ForwardRef onPress={[Function: onPress]} style={{...}}>
    <withTheme(TextComponent) style={{...}} />
  </ForwardRef>
  <ForwardRef onPress={[Function: onPress]} style={{...}}>
    <withTheme(TextComponent) style={{...}} />
  </ForwardRef>
  <ForwardRef onPress={[Function: onPress]} style={{...}}>
    <withTheme(TextComponent) style={{...}} />
  </ForwardRef>
</View>"
`;
